include/math/mls_polyinterpolation.tcc

```diff
@@ -0,0 +1,148 @@
+// Constructor
+template <unsigned Tdim, unsigned Tnmonomials>
+mpm::MLSPolyInterpolation<Tdim, Tnmonomials>::MLSPolyInterpolation()
+    : mpm::PolynomialInterpolation<Tdim>() {
+  M_matrix_.setZero();
+  B_vector_.clear();
+  weights_.clear();
+}
+
+// Initialise interpolator
+template <unsigned Tdim, unsigned Tnmonomials>
+void mpm::MLSPolyInterpolation<Tdim, Tnmonomials>::initialise(
+    const VectorDim& pcoord, const std::vector<VectorDim>& data_points,
+    unsigned spline_order, unsigned poly_order, double span) {
+
+  // Check spline order
+  if (spline_order < 2 || spline_order > 3)
+    throw std::runtime_error("MLS spline order is invalid");
+  // Check coordinates list of data points
+  if (data_points.empty())
+    throw std::runtime_error("MLS Data points coordinates are empty");
+  // Initialise monomial values of the polynomial at the given point
+  this->point_monomials_ =
+      mpm::Polynomial::evaluate_monomials<Tdim>(poly_order, pcoord);
+  // Compute weights
+  this->compute_weights(pcoord, data_points, spline_order, span);
+  // Initialise M matrix for the given set of points
+  this->initialise_M_matrix(data_points, poly_order);
+  // Initialise B vector for the given set of points
+  this->initialise_B_vector(data_points, poly_order);
+}
+
+// Compute associated weights of the data points set
+template <unsigned Tdim, unsigned Tnmonomials>
+void mpm::MLSPolyInterpolation<Tdim, Tnmonomials>::compute_weights(
+    const VectorDim& point, const std::vector<VectorDim>& data_points,
+    unsigned spline_order, double span) {
+
+  weights_.clear();
+  // Iterate over each data point and compute the associated weight
+  for (const auto& pdata : data_points) {
+    // Distance to data point
+    VectorDim distance;
+    for (unsigned dim = 0; dim < Tdim; ++dim)
+      distance(dim) = std::fabs(pdata(dim) - point(dim)) / span;
+    // Spline weight
+    double weight = std::numeric_limits<double>::max();
+    if (spline_order == 2) weight = this->quadratic_spline_weight(distance);
+    if (spline_order == 3) weight = this->cubic_spline_weight(distance);
```

Comment:
if the spline order is not available, wouldn't it be better to throw error rather than passing std::numeric_limits<double>::max(), which is wrong, I think.